src/main/java/ether/EtherClient.java

```diff
@@ -30,238 +29,280 @@ private EtherClient() {
-     * @throws DecoderException
-     * @throws IOException
+     * @param senderAddress         The sender address in the Ethereum Network
+     * @param receiverAddress       The receiver address in the Ethereum Network
+     * @param httpAgent             Http wrapper
+     * @return                      PaymentChannel object that holds all channel info.
+     * @throws IOException          Thrown if fails to get http response from a remote mode.
      */
     public static EtherUtility.PaymentChannel getChannelInfo(String senderAddress, String receiverAddress
-            , Http httpAgent) throws DecoderException, IOException {
+            , Http httpAgent) throws IOException {
 
         byte[] keyInBytes = EtherUtility.getChannelHash(senderAddress, receiverAddress);
-        if(keyInBytes == null){
+        if (keyInBytes == null) {
             if (Settings.DEBUG_INFO) {
-                System.out.println("Failed to construct the channel Hash: "+senderAddress+"-->"+receiverAddress);
+                System.out.println("Failed to construct the channel Hash: " + senderAddress + "-->" + receiverAddress);
             }
             return null;
         }
 
-        String requestString = getEtherCallRequest("channels", keyInBytes);
+        CallTransaction.Function func = Settings.CHANNEL_CONTRACT.getByName("channels");
+        byte[] funcBytes = func.encode(keyInBytes);
+        String functionBytesStr = "0x" + new String(Hex.encodeHex(funcBytes));
+        String request = getEtherRequest("eth_call", null,
+                Settings.CHANNEL_CONTRACT_ADDRESS, null, functionBytesStr);
 
         if (Settings.DEBUG_INFO) {
-            System.out.println("Request in getChannelInfo = " + requestString);
+            System.out.println("Request in getChannelInfo = " + request);
         }
 
-        String checkChannelAvailableResult;
+        String response;
         try {
-            checkChannelAvailableResult = (String) httpAgent.getHttpResponse(requestString);
+            response = (String) httpAgent.getHttpResponse(request);
         } catch (IOException e) {
             if (Settings.DEBUG_INFO) {
                 System.out.println("Cannot checkChannelAvailable from " + senderAddress + " to " + receiverAddress);
             }
             throw e;
         }
 
-        //TODO: decode channel info;
-        return new EtherUtility.PaymentChannel("0x111",
-                "0x222", Settings.MAX_DEPOSIT, 1, null, null);
+        if (response == null || response == "") {
+            return null;
+        }
+
+        if (Settings.DEBUG_INFO) {
+            System.out.println("checkChannelAvailable response: " + response);
+        }
+
+        response = response.substring(2);
+        BigInteger res = new BigInteger(response, 16);
+        if (res.equals(BigInteger.ZERO)) {
+            return null;
+        }
+
+        String depositStr = response.substring(0, response.length() / 2);
```

Comment:
What is the format of the response? Could you please add a comment for the reason of dividing by 2?

Reply:
The response contains the Channel struct, struct is returned as an array of bytes
Channel struct contains 2 members - initial deposit and the open block number.
I will add a comment in the code.